lib/src/__tests__/DatePickerTestingLib.test.tsx

```diff
@@ -1,24 +1,94 @@
-import * as React from 'react';
-import { DatePicker } from '../DatePicker';
+import React, { useState } from 'react';
+import frLocale from 'date-fns/locale/fr';
+import DateFnsUtils from '@date-io/date-fns';
+import enLocale from 'date-fns/locale/en-US';
+import Button from '@material-ui/core/Button';
 import TextField from '@material-ui/core/TextField';
-import { screen } from '@testing-library/react';
-import { createClientRender } from './createClientRender';
+import LocalizationProvider from '../LocalizationProvider';
+import { MaterialUiPickersDate } from '../typings';
+import { screen, fireEvent, render } from '@testing-library/react';
+import { DesktopDatePicker, DatePickerProps } from '../DatePicker/DatePicker';
 
-describe('<DatePicker />', () => {
-  const render = createClientRender({ strict: false });
-  it('Allows to select edge years from list', () => {
-    render(
-      <DatePicker
-        open
-        value={null}
-        onChange={jest.fn()}
-        views={['year']}
-        minDate={new Date('2000-01-01')}
-        maxDate={new Date('2010-01-01')}
-        renderInput={props => <TextField {...props} />}
+const localeMap = {
+  en: enLocale,
+  fr: frLocale,
+};
+
+interface FormProps {
+  picker: React.FC<DatePickerProps>;
+}
+
+const Form: React.FC<FormProps> = ({ picker: Picker }) => {
+  const [date, setDate] = useState<MaterialUiPickersDate>(new Date('01/01/2020'));
+  const [locale, setLocale] = useState('en');
+  return (
+    <LocalizationProvider dateAdapter={DateFnsUtils} locale={localeMap[locale]}>
+      <Picker
+        value={date}
+        onChange={setDate}
+        views={['year', 'month']}
+        renderInput={props => {
+          const inputProps = {
+            ...props.inputProps,
+            'aria-label': 'test input',
+          };
+          return <TextField {...props} inputProps={inputProps} />;
+        }}
       />
-    );
+      <div>
+        <Button onClick={() => setLocale('en')}>En</Button>
+        <Button onClick={() => setLocale('fr')}>Fr</Button>
+      </div>
+    </LocalizationProvider>
+  );
+};
+
+describe('<DatePicker />', () => {
+  it('Should add aria-invalid of true to input when date is invalid', () => {
+    render(<Form picker={DesktopDatePicker} />);
```

Comment:
We try to keep the tests as single focused and isolated as possible. In our case, do we need the locale switcher for this test? I don't think so, I think that we should go one level of abstraction down. 

Also, what about running the tests with each date engine. Is this something we should aim for, or is it overkill at this level? Maybe the former?

Reply:
Agreed regarding each date engine.  Hijri is not validating correctly with the views={['month', 'year']}.  Maybe that calendar requires a different format. Jalali validates correctly, however, when the input is changed to invalid text and then changed back to valid text the field remains invalid.

Thoughts on Hijri and Jalali?